### PR TITLE
MM-11227: Fix group msg slash command.

### PR DESCRIPTION
**Pull request by grundleborg**
#### Summary
Fix group msg slash command.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-11227

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Includes text changes and localization file updates

### CODE REVIEW

File: app/channel.go [hunk 1]
```diff
@@ -340,6 +340,30 @@ func (a *App) createGroupChannel(userIds []string, creatorId string) (*model.Cha
 	}
 }
 
+func (a *App) GetGroupChannel(userIds []string) (*model.Channel, *model.AppError) {
+	if len(userIds) > model.CHANNEL_GROUP_MAX_USERS || len(userIds) < model.CHANNEL_GROUP_MIN_USERS {
```
Comment (jespino) [inline app/channel.go]: Do we need this check?

File: app/channel.go [hunk 2]
```diff
@@ -340,6 +340,30 @@ func (a *App) createGroupChannel(userIds []string, creatorId string) (*model.Cha
...
+	}
+
+	var users []*model.User
+	if result := <-a.Srv.Store.User().GetProfileByIds(userIds, true); result.Err != nil {
+		return nil, result.Err
+	} else {
+		users = result.Data.([]*model.User)
+	}
+
+	if len(users) != len(userIds) {
```
Comment (jespino) [inline app/channel.go]: And this check?

File: app/command_groupmsg_test.go [hunk 3]
```diff
@@ -35,3 +40,58 @@ func TestGroupMsgUsernames(t *testing.T) {
...
+
+	user3 := th.CreateUser()
+	targetUsers := "@" + th.BasicUser2.Username + ",@" + user3.Username + " "
+
+	team := th.CreateTeam()
+	th.LinkUserToTeam(th.BasicUser, team)
+	cmd := &groupmsgProvider{}
+
+	// Check without permission to create a GM channel.
+	resp := cmd.DoCommand(th.App, &model.CommandArgs{
```
Comment (jespino) [inline app/command_groupmsg_test.go]: Maybe we can use here a Test Table

File: app/channel.go (hunk 1 above)
Comment (jespino) [reply app/channel.go]: Ok, my point here is, if you can't create a GroupChannel with that limits, the channel wont exists anyway, so, you can try to get it and will fail as "not-exists" more proper error than, "you can't create it"

File: app/channel.go (hunk 2 above)
Comment (jespino) [reply app/channel.go]: Something similar here, if you can't create the group channel with a not existing users, a group channel must have all the users. The exception would be the permanent deletion of a user. With this check If any user is permanently deleted from the database, it won't allow you to access to the group channel. Without this check, you will be able to see it (at least, through the API).

Comment (grundleborg) [reply app/channel.go]: OK, so I take your point, but I don't want to change the existing behaviour in this PR, so I'm going to leave this as-is to be consistent with the `CreateGroupChannel` function that it is being used alongside (and is copy-pasted from). If we want to fix the issue you've highlighted, it should be a separate ticket.

Comment (jespino) [reply app/channel.go]: Fair enough

File: app/channel.go (hunk 1 above)
Comment (grundleborg) [reply app/channel.go]: OK. However, given we then fetch all the users requested from the database, I'd prefer to filter out excess requests before we hit the database with them. Also, although the error "key" mentions `create`, the actual string is suitably generic for this case as well.

**Pull request merged**